lib/src/presentation/pages/movie_details.dart

```diff
@@ -1,43 +1,72 @@
 import 'package:flutter/material.dart';
+
+import '../../core/util/enums.dart';
 import '../../core/util/ui_consts.dart';
+import '../../domain/entity/cast.dart';
+import '../../domain/entity/event.dart';
 import '../../domain/entity/movie.dart';
+import '../bloc/cast_bloc.dart';
+import '../bloc/movie_bloc.dart';
+import '../widgets/loaders/cast_loader.dart';
+import '../widgets/movie/cast_widget.dart';
+import '../widgets/movie/movie_details_custom_nav_bar.dart';
 import '../widgets/movie/movie_presentation.dart';
 import '../widgets/movie/overview.dart';
-import '../widgets/movie/movie_details_custom_nav_bar.dart';
-import '../widgets/movie/cast.dart';
+import '../widgets/shared/error_class.dart';
 
-class MovieDetails extends StatefulWidget {
-  const MovieDetails({
-    super.key,
-    required this.movie,
-  });
+class MovieDetails extends StatelessWidget {
+  static const double endPadding = 130;
+  static const String message = 'Error to bring the cast';
 
   final Movie movie;
-  static const String routeName = '/movie';
+  final MovieBloc movieBloc;
+  final CastBloc castBloc;
 
-  @override
-  State<MovieDetails> createState() => _MovieDetailsState();
-}
-
-class _MovieDetailsState extends State<MovieDetails> {
-  static const double endPadding = 130;
+  const MovieDetails({
+    required this.movie,
+    required this.movieBloc,
+    required this.castBloc,
+    super.key,
+  });
 
   @override
   Widget build(BuildContext context) {
+    movieBloc.setLastMovie(movie);
+    castBloc.loadCast(movie);
```

Comment:
Please make this widget `Stateful` and call these methods in the `initState` method

Reply:
Okay, done it